src/Events/Admin/Onboarding/Controller.php

```diff
@@ -88,6 +88,53 @@ public function handle_onboarding_page_dismiss(): void {
 		$this->container->make( Landing_Page::class )->handle_onboarding_page_dismiss();
 	}
 
+	/**
+	 * Redirects users to the Guided Setup page when accessing any TEC settings or management page for the first time.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function redirect_tec_pages_to_guided_setup(): void {
+		// Get some information about the page that we are trying to access.
+		$page      = tribe_get_request_var( 'page' );
+		$post_type = tribe_get_request_var( 'post_type' );
+
+		// Do not redirect if they have older versions and are probably already set up.
+		$tec_versions = (array) tribe_get_option( 'previous_ecp_versions', [] );
+		if ( count( $tec_versions ) > 1 ) {
+			return;
+		}
+
+		// Do not redirect if the target is not The Events Calendar-related admin pages.
+		if ( ! str_contains( $post_type, 'tribe' ) ) {
+			return;
+		}
+
+		// Do not redirect if they are already on the Guided Setup page.
+		if ( 'first-time-setup' === $page ) {
+			return;
+		}
+
+		// Do not redirect if they have been to the Guided Setup page already.
+		$visited_first_time_setup = tribe_get_option( 'tec_onboarding_wizard_visited_guided_setup', false );
+		if ( $visited_first_time_setup ) {
+			return;
+		}
+
+		// If we're still here, redirect to the Guided Setup page.
+		$setup_url = add_query_arg(
+			[
+				'post_type' => 'tribe_events',
+				'page'      => 'first-time-setup',
+			],
+			admin_url( 'edit.php' )
+		);
+
+		wp_safe_redirect( $setup_url );
+		exit;
```

Comment:
```suggestion
		tribe_exit();
```

We dont use exit or die in our code so that we can test the methods safer.